grpc/src/test/java/com/linecorp/armeria/server/grpc/GrpcServiceImplErrorTest.java

```diff
@@ -0,0 +1,319 @@
+            });
+            assertThat(log.responseCause())
+                    .isInstanceOf(StatusRuntimeException.class)
+                    .satisfies(cause -> {
+                        assertThat(Status.fromThrowable(cause).getCode()).isEqualTo(
+                                Status.INTERNAL.getCode());
+                        assertThat(Status.trailersFromThrowable(cause)).satisfies(metadata -> {
+                            assertThat(isSameSizeAndNameAndValues(metadata, corruptedMetadata)).isFalse();
+                        });
+                    });
+        }
+    }
+
+    private static final class ServerImplErrorAtMetadataService extends TestServiceGrpc.TestServiceImplBase {
+        // blocking client
+        // No error happens when serializing metadata.
+        @Override
+        public void unaryCall2(SimpleRequest request, StreamObserver<SimpleResponse> responseObserver) {
+            responseObserver.onError(new StatusRuntimeException(Status.ABORTED, validMetadata));
+        }
+
+        // blocking client
+        // Error happens when serializing metadata.
+        @Override
+        public void unaryCall(SimpleRequest request, StreamObserver<SimpleResponse> responseObserver) {
+            responseObserver.onError(new StatusRuntimeException(Status.ABORTED, corruptedMetadata));
+        }
+
+        // blocking client
+        // Error happens when serializing metadata.
+        @Override
+        public void streamingOutputCall(StreamingOutputCallRequest request,
+                                        StreamObserver<StreamingOutputCallResponse> responseObserver) {
+            responseObserver.onNext(StreamingOutputCallResponse.newBuilder().setPayload(
+                    Payload.newBuilder().setBody(ByteString.copyFromUtf8("pong1")).build()).build());
+            if (request.equals(STREAMING_OUTPUT_CALL_REQUEST_MESSAGE)) {
+                responseObserver.onError(
+                        new StatusRuntimeException(Status.FAILED_PRECONDITION, corruptedMetadata));
+                return; // onError must be the last call.
+            }
+            // The following lines are not reached.
+            responseObserver.onNext(StreamingOutputCallResponse.newBuilder().setPayload(
+                    Payload.newBuilder().setBody(ByteString.copyFromUtf8("pong2")).build()).build());
+            responseObserver.onCompleted();
+        }
+    }
+
+    private static void configureGrpcService(
+            ServerBuilder sb, ServerInterceptor... interceptors) {
+        sb.service(GrpcService.builder().addService(new ServerImplErrorAtMetadataService())
+                              .intercept(interceptors).build());
+        sb.decorator(LoggingService.newDecorator());
+    }
+
+    // Use this method to ensure to escape assertion in the constructor of Metadata.
+    private static boolean isSameSizeAndNameAndValues(Metadata a, Metadata b) {
```

Comment:
Yes.
You noticed good points.
Current commit is useless, so it's better to be deleted.

I had thought that it might be better to create `this-metadata-is-not-contained` method instead.